README.adoc

```diff
@@ -2,25 +2,151 @@
+include::name-sink/src/main/resources/application.properties[tag=sinkexchangeconfig]
+----
+====
+
+
+To test the sink we will manually publish a JSON message representing a `Person` record to the newly generated exchange:
+
+image::rabbitmq-sink.png[]
+
+And in our application logs we should see the data outputted:
+
+image::output-sink.png[]
+
+
+
+== Processor Application
+
+The processor (`java.util.function.Function`) is defined in `NameProcessorConfiguration` as
+
+====
+[source,java,tabsize=2,indent=0]
+----
+include::name-processor/src/main/java/com/example/stream/nameprocessor/NameProcessorConfiguration.java[tag=processname]
+----
+====
+
+This function will take a String value as input and create a new Person record that adds the timestamp of when the data was processed.
+Running this application will create 2 new exchanges in RabbitMQ: `processName-in-0` and `processName-out-0`.
+Similar to the configuration we applied to the sink application, we want to change these exchange names so they can hook into the
+sink and soon to the supplier.
+
+====
+[source,tabsize=2,indent=0]
+----
+include::name-processor/src/main/resources/application.properties[tag=processorexchangeconfig]
+----
+====
+
+NOTE: The output of the processor matches the input of the sink.
+
+Using the RabbitMQ dashboard, we will now send a String (name) to the processor input exchange and watch as it flows to the connected sink.
+
+image::rabbitmq-processor.png[]
+
+If the processor and the sink are connected properly, you should still the output from your running sink:
+
+image::output-processor-sink.png[]
+
+== Source Application
+
+
+The Source (`java.util.function.Supplier`) is defined in `NameSourceConfiguration` as
+
+====
+[source,java,tabsize=2,indent=0]
+----
+include::name-source/src/main/java/com/example/stream/namesource/NameSourceConfiguration.java[tag=supplyname]
+----
+====
 
-link:https://dataflow.spring.io/docs/stream-developer-guides/streams/standalone-stream-sample/[Stream Processing with Kafka]
+Similar to how we hooked up the processor output to the sink input, we must do the same and connect the source output to the
+processor input.
 
-=== Stream Processing with RabbitMQ
-In the following guide, we develop three Spring Boot applications that use Spring Cloud Stream's support for RabbitMQ and deploy
-them to Cloud Foundry, to Kubernetes, and on your local machine.
+====
+[source,tabsize=2,indent=0]
+----
+include::name-source/src/main/resources/application.properties[tag=supplierexchangeconfig]
+----
+====
 
-_(There is a single guide for both Apache Kafka and RabbitMQ - select the 'RabbitMQ' tabs)_.
+NOTE: The output of the source should matches the input of the processor
 
-link:https://dataflow.spring.io/docs/stream-developer-guides/streams/standalone-stream-sample/[Stream Processing with RabbitMQ]
+If `name-processor` and `name-sink` are already running, starting the `name-source` will immediately start messages flowing through the system.
```

Comment:
Change "will immediately start" to "immediately starts". (In other words, remove "will".